src/AppInstallerCommonCore/Public/winget/ManifestCommon.h

```diff
@@ -112,20 +112,58 @@ namespace AppInstaller::Manifest
         Preview,
     };
 
-    struct PackageDependency
+    enum class DependencyType
     {
-        string_t Id;
-        string_t MinVersion;
+        WindowsFeature,
+        WindowsLibraries,
+        Package,
+        External
     };
 
     struct Dependency
     {
-        std::vector<string_t> WindowsFeatures;
-        std::vector<string_t> WindowsLibraries;
-        std::vector<PackageDependency> PackageDependencies;
-        std::vector<string_t> ExternalDependencies;
+        DependencyType Type;
+        string_t Id;
+        std::optional<string_t> MinVersion;
+
+        Dependency(DependencyType type, string_t id, std::optional<string_t> minVersion) : Type(type), Id(id), MinVersion(minVersion) {}
+        Dependency(DependencyType type, string_t id) : Type(type), Id(id) {}
+        Dependency(DependencyType type) : Type(type) {}
     };
 
+    struct DependencyList
+    {
+        std::vector<Dependency> dependencies;
+
+        DependencyList() {}

```

Comment:
If you want a default behavior for constructors et al., use `DependencyList() = default;`